src/v/cloud_storage/partition_manifest.h

```diff
@@ -80,6 +87,12 @@ class partition_manifest final : public base_manifest {
 
         model::initial_revision_id ntp_revision;
         model::term_id archiver_term;
+        /// Term of the segment (included in segment file name)
+        model::term_id segment_term;
+        /// Offset translation delta at the end of the range
+        model::offset_delta delta_offset_end;
+        /// Version (used by JSON deserializer)
+        segment_meta_version version{segment_meta_version::v1};
```

Comment:
I don't particularly like calling this "version" (easy to confuse with the serde version). AFAICT it only affects how the remote segment name is constructed (and the comment is wrong, json deserializer doesn't use it)? Maybe call it `segment_name_format` or just trigger the format change if `delta_offset_end` is valid (though this is hacky...)?